azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_stages_provisioning_mqtt.py

```diff
@@ -43,12 +44,14 @@ def _run_op(self, op):
             # always pass it down because MQTT protocol stage will also want to receive these args.
 
             client_id = op.registration_id
-            client_version = urllib.parse.quote_plus(constant.USER_AGENT)
-            username = "{id_scope}/registrations/{registration_id}/api-version={api_version}&ClientVersion={client_version}".format(
+            query_params = [
+                ("api-version", pkg_constant.PROVISIONING_API_VERSION),
+                ("ClientVersion", pkg_constant.USER_AGENT),
+            ]
+            username = "{id_scope}/registrations/{registration_id}/?{query_params}".format(
```

Comment:
>? [](start = 67, length = 1)

as we found out yesterday that this question mark is not needed for DPS.  To maintain similarity with other SDKs we should remove this. But it is better to add a Note/Comment above.

Reply:
I'm going to wait until I get an answer from Robledo regarding whether or not the DPS hub prefers this.

---
In reply to: [307828656](https://github.com/Azure/azure-iot-sdk-python-preview/pull/147#discussion_r307828656) [](ancestors = 307828656)